allensdk/brain_observatory/behavior/session_apis/data_io/behavior_lims_api.py

```diff
@@ -0,0 +1,298 @@
+            JOIN genders g ON g.id = d.gender_id
+            WHERE bs.id = {self.behavior_session_id};
+            """
+        return self.lims_db.fetchone(query, strict=True)
+
+    @memoize
+    def get_age(self) -> str:
+        """Returns age code of the subject.
+        :rtype: str
+        """
+        query = f"""
+            SELECT a.name AS age
+            FROM behavior_sessions bs
+            JOIN donors d ON d.id = bs.donor_id
+            JOIN ages a ON a.id = d.age_id
+            WHERE bs.id = {self.behavior_session_id};
+        """
+        return self.lims_db.fetchone(query, strict=True)
+
+    @memoize
+    def get_rig_name(self) -> str:
+        """Returns the name of the experimental rig.
+        :rtype: str
+        """
+        query = f"""
+            SELECT e.name AS device_name
+            FROM behavior_sessions bs
+            JOIN equipment e ON e.id = bs.equipment_id
+            WHERE bs.id = {self.behavior_session_id};
+        """
+        return self.lims_db.fetchone(query, strict=True)
+
+    @memoize
+    def get_stimulus_name(self) -> str:
+        """Returns the name of the stimulus set used for the session.
+        :rtype: str
+        """
+        query = f"""
+            SELECT stages.name
+            FROM behavior_sessions bs
+            JOIN stages ON stages.id = bs.state_id
+            WHERE bs.id = '{self.foraging_id}'
+        """
+        return self.mtrain_db.fetchone(query, strict=True)
+
+    @memoize
+    def get_reporter_line(self) -> List[str]:
+        """Returns the genotype name(s) of the reporter line(s).
+        :rtype: list
+        """
+        query = f"""
+            SELECT g.name AS reporter_line
+            FROM behavior_sessions bs
+            JOIN donors d ON bs.donor_id=d.id
+            JOIN donors_genotypes dg ON dg.donor_id=d.id
+            JOIN genotypes g ON g.id=dg.genotype_id
+            JOIN genotype_types gt
+                ON gt.id=g.genotype_type_id AND gt.name = 'reporter'
+            WHERE bs.id={self.behavior_session_id};
+        """
+        result = self.lims_db.fetchall(query)
+        if result is None or len(result) < 1:
+            raise OneOrMoreResultExpectedError(
+                f"Expected one or more, but received: '{result}' "
+                f"from query:\n'{query}'")
+        return result
+
+    @memoize
+    def get_driver_line(self) -> List[str]:
+        """Returns the genotype name(s) of the driver line(s).
+        :rtype: list
+        """
+        query = f"""
+            SELECT g.name AS driver_line
+            FROM behavior_sessions bs
+            JOIN donors d ON bs.donor_id=d.id
+            JOIN donors_genotypes dg ON dg.donor_id=d.id
+            JOIN genotypes g ON g.id=dg.genotype_id
+            JOIN genotype_types gt
+                ON gt.id=g.genotype_type_id AND gt.name = 'driver'
+            WHERE bs.id={self.behavior_session_id};
+        """
+        result = self.lims_db.fetchall(query)
```

Comment:
`fetchall` doesn't actually use the strict parameter.

Reply:
Turns out there is a lot of weirdness with: https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/internal/api/__init__.py

I'm hesitant to touch this for now as it could have sweeping repercussions.